src/runtime/scheduler_cortexm.S

```diff
@@ -111,3 +111,35 @@ tinygo_swapTask:
     mov r11, r3
     pop {pc}
     #endif
+
+.global  tinygo_scanCurrentStack
+.type    tinygo_scanCurrentStack, %function
+tinygo_scanCurrentStack:
+    // Save callee-saved registers onto the stack.
+    #if defined(__thumb2__)
+    push {r4-r11, lr}
+    #else
+    mov r0, r8
+    mov r1, r9
+    mov r2, r10
+    mov r3, r11
+    push {r0-r3, lr}
+    push {r4-r7}
+    #endif
+
+    // Scan the stack.
+    mov r0, sp
+    bl tinygo_scanstack
+
+    // Reload callee-saved registers and return.
+    #if defined(__thumb2__)
+    pop {r4-r11, pc}
+    #else
+    pop {r4-r7}
+    pop {r0-r3}
+    mov r8, r0
+    mov r9, r1
+    mov r10, r2
+    mov r11, r3
+    pop {pc}
+    #endif
```

Comment:
You should be able to replace this with the following:

```
        add     sp, #32
        pop     {pc}
```

That's one instruction more on thumb2 but a whole lot faster in all cases.

Reply:
Done.